src/app/wizard/services/http.service.ts

```diff
@@ -1,60 +1,69 @@
-import { Location } from "@angular/common";
-import { Headers, Http, RequestOptions, Response } from '@angular/http';
-import { Observable } from "rxjs";
-import { HelperService, TokenProvider, Cluster } from "ngx-forge";
+import { Location } from '@angular/common';
+import { HttpClient, HttpErrorResponse, HttpHeaders } from '@angular/common/http';
+import { Cluster, HelperService, TokenProvider } from 'ngx-forge';
+import { catchError, flatMap } from 'rxjs/operators';
+import { Observable } from 'rxjs-compat';
+import { throwError } from 'rxjs';
+import { fromPromise } from 'rxjs-compat/observable/fromPromise';
 
 export class HttpService {
+
+  public static handleError(error: HttpErrorResponse) {
+    if (error.error instanceof ErrorEvent) {
+      // A client-side or network error occurred. Handle it accordingly.
+      console.error('An error occurred:', error.error.message);
+    } else {
+      // The backend returned an unsuccessful response code.
+      // The response body may contain clues as to what went wrong,
+      console.error(
+        `Backend returned code ${error.status}, ` +
+        `body was: ${error.error}`);
+    }
+    // return an observable with a user-facing error message
+    return throwError(
+      'Something bad happened; please try again later.');
+  }
+
   constructor(
-    private http: Http,
+    private http: HttpClient,
     private helperService: HelperService,
     private tokenProvider: TokenProvider
   ) { }
 
-  httpGet<T>(...endpoint: string[]): Observable<T> {
+  public backendHttpGet<T>(...endpoint: string[]): Observable<T> {
     endpoint.unshift(this.helperService.getBackendUrl());
-    return this.options().flatMap((option) => {
-      const url = this.joinPath(endpoint);
-      return this.http.get(url, option)
-        .map(response => response.json() as T)
-        .catch(this.handleError);
-    });
+    return this.options().pipe(
+        flatMap((options) => this.http.get<T>(this.joinPath(...endpoint), options)),
+        catchError(HttpService.handleError)
+      );
   }
 
-  protected joinPath(parts: string[]): string {
+  protected joinPath(...parts: string[]): string {
+    if (!parts || parts.length === 0) {
+      return '';
+    }
     let result = parts[0];
-    for (let i = 1; i < parts.length; i++) {
-      result = Location.joinWithSlash(result, parts[i]);
+    if (parts.length === 1) {
```

Comment:
this check is not nessary if parts.length === 1 the loop will not start

Reply:
indeed :) `['toto'].slice(1) => []`